scripts/zones/Sacrarium/mobs/Mariselles_Pupil.lua

```diff
@@ -0,0 +1,132 @@
+        { 88.3052, -2, 118.776 },
+        { 88.8908, -2.1785, 129.5286 },
+        { 100.0305, -2.7298, 130.7616 },
+        { 110.8072, -2.1785, 130.0085 },
+        { 100.0483, -2, 116.4553 },
+        },
+        --Northmid room
+        [1] =
+        {
+        { 70.8789, -2, 109.542 },
+        { 60.4267, -2, 110.0887 },
+        { 48.2896, -2, 111.3439 },
+        { 49.2171, -2, 117.5056 },
+        { 61.5946, -2, 116.0528 },
+        { 71.7462, -2, 116.0356 },
+        { 69.0619, -2.1785, 130.0243 },
+        { 58.1095, -2.7298, 129.9273 },
+        { 50.1262, -2.1785, 130.2446 },
+        },
+        --Northwest room
+        [2] =
+        {
+        { 9.7012, -2, 109.5587 },
+        { 22.042, -2, 110.0485 },
+        { 31.9411, -2, 112.7432 },
+        { 30.6844, -2, 117.8574 },
+        { 18.2341, -2, 116.4862 },
+        { 7.4438, -2, 116.2394 },
+        { 9.3145, -2.1785, 130.6581 },
+        { 19.8528, -2.7298, 130.7574 },
+        { 30.7859, -2.1785, 129.5448 },
+        },
+        --Southeast room
+        [3] =
+        {
+        { 110.3006, -2.1785, -129.8003 },
+        { 98.4509, -2.7298, -129.8176 },
+        { 90.0133, -2.1785, -128.4242 },
+        { 89.6509, -2.0000, -115.7570 },
+        { 100.2014, -2.0000, -115.8586 },
+        { 112.5715, -2.0000, -115.8909 },
+        { 111.0784, -2.0000, -109.7074 },
+        { 97.4817, -2.0000, -109.9677 },
+        { 88.0128, -2.0000, -110.1289 },
+        },
+        --Southmid room
+        [4] =
+        {
+        { 49.0167, -2.0000, -109.8401 },
+        { 60.2168, -2.0000, -109.9171 },
+        { 73.0027, -2.0000, -110.6675 },
+        { 72.0441, -2.0000, -115.7637 },
+        { 58.3822, -2.0000, -116.0210 },
+        { 48.5663, -2.0000, -116.0103 },
+        { 49.0839, -2.1785, -130.2026 },
+        { 62.3159, -2.6880, -129.7956 },
+        { 69.3685, -2.1785, -129.3329 },
+        },
+        --Southwest room
+        [5] =
+        {
+        { 8.7566, -2.0000, -109.7524 },
+        { 20.8865, -2.0000, -110.3810 },
+        { 31.4743, -2.0000, -110.4831 },
+        { 31.4395, -2.0000, -115.9673 },
+        { 18.1959, -2.0000, -116.3700 },
+        { 8.1382, -2.0000, -116.6599 },
+        { 11.1776, -2.1785, -129.9487 },
+        { 19.9970, -2.7298, -129.7488 },
+        { 29.3828, -2.1785, -129.9017 },
+        }
+    }
+
+    local teleTime = mob:getLocalVar("teleTime")
+    if mob:getBattleTime() - teleTime > 30 and mob:getBattleTime() > 59 and mob:actionQueueEmpty() == true then
+        local profLocation = mob:getLocalVar("spawnLocation")
+        local randomPosition = math.random(1, 9)
+        utils.mobTeleport(mob, 2000, Locations[profLocation][randomPosition])
+        mob:setLocalVar("teleTime", mob:getBattleTime())
+    end
+end
+
+entity.onMobDisengage = function(mob)
+    mob:setLocalVar("teleTime", 0)
+end
+
+entity.onMobRoam = function(mob)
+    local profLocation = mob:getLocalVar("spawnLocation")
+    local returnPoint =
+    {
```

Comment:
Indents on the table, but also could this be formatted to make the values align vertically?

Reply:
Extra spaces in the table to align things? Other than that fixed

Reply:
I've committed what I THINK you want.